airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/factory.py

```diff
@@ -52,6 +53,8 @@ def _create_subcomponent(self, v, kwargs, config):
             v["options"] = self._merge_dicts(kwargs.get("options", dict()), v.get("options", dict()))
 
             return self.create_component(v, config)()
+        elif isinstance(v, dict):
+            return InterpolatedMapping(v).eval(config=config, kwargs=kwargs)
```

Comment:
on second thought, this introduces kind of a concerning side effect to our component generation. For most of our existing components, we interpolate at run time, this change to the factory forces it at parse/instantiation time for _all_ components, which we don't want...

I think this all comes about since we're reusing the existing `Oauth2Authenticator` shared by declarative and normal integrations. I'd have to shove an interpolator and configs/kwargs/options into the class for `get_refresh_request_body()` to interpolate like I'd expect it to. I'll think on this a little bit more.

Reply:
@brianjlai would it make sense for the factory to create an InterpolatedMapping instead of a standard map when there is no `class_name` so we can always do interpolation?